### ISSUE TITLE
Why RMSE shows better results than R2 for the loss function?

### ISSUE THREAD

**Issue by geek-yang** [Kag-Jane/sandbox#5]
It seems we are not the only one....Check [this post](https://www.kaggle.com/competitions/jane-street-real-time-market-data-forecasting/discussion/550453#3069518). Here is what it says:

>> Loss function: I've tinkered with the loss function quite a bit. I found that an r2 loss can lead to the local minimum of predicting near zero for every instance (e.g. the trivial minimum) due in part to the sum of true labels being in the denominator. If you truly want to maximize the LB metric, your model should be generating predictions which minimize the term in the numerator of the r2 score. MSE, RMSE, and MAE have all worked well for me. I have not played around with including weights in the loss, but my intuition tells me that this would cause issues based on the amount of noise in the data.

It also suggests to not include the weight. The post below indicates the same thing:
>> remove weights from loss_fn during training
